src/mondo_robot_template.py

```diff
@@ -34,13 +34,10 @@ def _prefixed_id_rows_from_common_df(source_df: pd.DataFrame, mondo_col='mondo_i
     df = copy(source_df)
     df[xref_col] = df[xref_col].apply(
         lambda x: f'MEDGENCUI:{x}' if x.startswith('CN')  # "CUI Novel"
-        else f'UMLS:{x}' if x.startswith('C')  # CUI 1 of 2: UMLS
+        else f'UMLS:{x}' if x.startswith('C')  # UMLS CUI
         else f'MEDGEN:{x}')  # UID
     rows = df.to_dict('records')
-    # CUI 2 of 2: MEDGENCUI:
-    rows2 = [{mondo_col: x[mondo_col], xref_col: x[xref_col].replace('UMLS', 'MEDGENCUI')} for x in rows if
-             x[xref_col].startswith('UMLS')]
```

Comment:
## Removed only 1 of 2 kinds of MEDGENCUI
- [ ] @matentzn Review and inform of determination

@matentzn Pertaining to step _Remove all rows with MEDGENCUI in it (we only need xrefs to MEDGEN:123 and UMLS:123)_ (#15), I just want to be sure about this. I think we may lose some mappings if we do this. There are 2 ways in which we currently get MEGENCUI. It's been a while, and I couldn't remember why we do this assignment in 2 ways instead of just 1. If we have the reason documented, it's not obvious to me where that is. Hmmm... perhaps we added it to the MedGen docs in `mondo-ingest` before we removed that file.

1. Via duplication from UMLS namespace
```py
    rows2 = [{mondo_col: x[mondo_col], xref_col: x[xref_col].replace('UMLS', 'MEDGENCUI')} for x in rows if
             x[xref_col].startswith('UMLS')]
    return rows + rows2
```
I understand the logic more in this.
This also came up in the HPO SSSOM PR ([comment](https://github.com/monarch-initiative/medgen/pull/18/files#r1536737518)), though currently there are only UMLS mappings there.

2. From all MedGen IDs that start with `CN` and not `C`
There are many cases of this, and these are not duplicative.
For more information/documentation on C/CN and our prefix assignemtn, see [this comment](https://github.com/monarch-initiative/medgen/pull/18/files#r1536688641).

### The question
Do you mean to remove MEDGENCUI as derived from both cases (1) and (2)? Completely? Or just (1)?

Reply:
> perhaps we added it to the MedGen docs in mondo-ingest before we removed that file.

As an aside: You could familiarise with search git logs: https://gun.io/news/2017/02/git-search-log/

> Do you mean to remove MEDGENCUI as derived from both cases (1) and (2)? Completely? Or just (1)?

The key is that we do not want MEDGENCUIs _at all_ in Mondo, regardless of where they come from.

I am assuming we are only "loosing" case 2 - case 1 is still there in the form of `UMLS` references.

Thanks for checking.

Reply:
@matentzn Thanks. I'll check out `git-search-log`.

I think you understand, but you stated it oppositely. I removed case 1 _Via duplication from UMLS namespace_, but left in **theoretically future possible** case 2 _From all MedGen IDs that start with CN and not C_. 

I might be being too detail-obsessed here, but I'd urge you to reconsider what you're asking me to do, and will hold off further code changes on this for the moment. There are no current case (2) CN<ID> to Mondo mappings, but in the future if they did add one, wouldn't we want to know about them / consider if they are valid? My current solution solves the problems of (i) getting rid of all MEDGENCUI instances now, because they are all of case (1) and (ii) leaves case (2) direct CN mappings open in the future so that we can formally evaluate if that contingency  arises, though I suppose that unless I throw an error if these arise, they could slip through without us realizing.

Reply:
TMD for me - you make a call, or ask Trish to make one. I only know one thing: whatever you do, _CNs should not make it into the robot template under no circumstances_ - neither now nor in the future (I mean, not enter _without a discussion_). They are an internal construct in MedGen that is 100% reflected by the MEDGEN:123 ids. Their use case is to create temporary UMLS IDs when a disease need to be curated that does not have am UMLS ID yet. You can leave the code if you like, but make sure that whatever happens they dont end up in the robot template (because that robot template is _merged blindly_, i.e. without review).

Reply:
Really glad to have your determination on this. Trish and I also discussed, and made:
- #20 

But it looks like the urgency is such that I should complete this issue within scope of this PR. So I will do it now.